src/Symfony/Component/DependencyInjection/Compiler/ResolveBindingsPass.php

```diff
@@ -171,6 +171,23 @@ protected function processValue($value, $isRoot = false)
                     continue;
                 }
 
+                if (0 !== \count($this->searchBindingByPropertyName($bindings, $parameter->name))) {
```

Comment:
the whole check should be moved after the `array_key_exists` below, isn't it?

Reply:
It is right now :)

`if (\array_key_exists($k = ltrim($typeHint, '\\').' $'.$parameter->name, $bindings)) `
on line 168 and than my condition on line 174

Reply:
I understood what you mean, but I think 
` if ($binding = $this->searchBindingByPropertyName($bindings, $parameter->name)) {
`
is enough